### ISSUE TITLE
semiautovivification

### ISSUE THREAD

**Issue by satyr** [satyr/coco#48]
```
a.b   ||= {}
a.b.c ||= []
a.b.c.push e
```

↓
    a%.b@.c.push e  # ((a.b ||= {})c ||= [])push e

Useful, or ugly?

**Comment by michaelficarra** [comment]
Both. Useful, but (syntactically) _very_ ugly.

**Comment by satyr** [comment]
> (syntactically) _very_ ugly

Perhaps. `%` and `@` make sense only when you know Perl sigils.

Can you think of better syntaxes?

**Comment by michaelficarra** [comment]
Not really, no.

**Comment by satyr** [comment]
Take 2

```
x . ! y . @ z  # ((x.y ||= {}).z ||= [])
```

which could be written as

```
x!y@z
```

thanks to ADI.

**Comment by michaelficarra** [comment]
Remember, by committing to that, you're disallowing future use of `!` and `@` as binary operators.

**Comment by satyr** [comment]
> by committing to that, you're disallowing future use of `!` and `@` as binary operators

Not really because of `.` prefix.

Remember though, the same argument applies to every new syntax. ;-)

**Comment by michaelficarra** [comment]
Thanks to ADI, you are. But it's true, that is a quality of every new syntax.

**Comment by satyr** [comment]
closed by 0f293ff266f109eaa0e17698ccc503d3b437bff9; implemented semiautovivification
